saleor/payment/gateways/adyen/webhooks.py

```diff
@@ -468,8 +470,14 @@ def validate_auth_user(headers: HttpHeaders, gateway_config: "GatewayConfig") ->
     return False
 
 
-def handle_webhook(request: WSGIRequest, gateway_config: "GatewayConfig"):
-    json_data = json.loads(request.body)
+def handle_webhook(
+    request: WSGIRequest, gateway_config: "GatewayConfig", payment_details: Callable
+):
+    try:
+        json_data = json.loads(request.body)
+    except ValueError:
+        return handle_additional_actions(request, payment_details)
```

Comment:
Not sure of this. 
Inside AdyenGatewayPlugin.webhook you can create simple `if` for separating webhook notification and additional actions. 
You have a request object so you can use request.path to define if this request should be treated as webhook notification or as a payment additional action. 
```
def webhook(request):
    if reguest.path == WEBHOOK_PATH:
        return handle_webhook()
    elif request.path == ADDITIONAL_PATH:
       return handle_additional_path()
```